torchvision/transforms/functional_tensor.py

```diff
@@ -48,3 +48,74 @@ def crop(img, top, left, height, width):
         raise TypeError('tensor is not a torch image.')
 
     return img[..., top:top + height, left:left + width]
+
+
+def adjust_brightness(img, brightness_factor):
+    """Adjust brightness of an RGB image.
+
+    Args:
+        img (Tensor): Image to be adjusted.
+        brightness_factor (float):  How much to adjust the brightness. Can be
+            any non negative number. 0 gives a black image, 1 gives the
+            original image while 2 increases the brightness by a factor of 2.
+
+    Returns:
+        Tensor: Brightness adjusted image.
+    """
+    if not F._is_tensor_image(img):
+        raise TypeError('tensor is not a torch image.')
+
+    black_img = torch.zeros(img.shape, device=img.device)
+
+    return _blend(img, black_img, brightness_factor)
+
+
+def adjust_contrast(img, contrast_factor):
+    """Adjust contrast of an RGB image.
+
+    Args:
+        img (Tensor): Image to be adjusted.
+        contrast_factor (float): How much to adjust the contrast. Can be any
+            non negative number. 0 gives a solid gray image, 1 gives the
+            original image while 2 increases the contrast by a factor of 2.
+
+    Returns:
+        Tensor: Contrast adjusted image.
+    """
+    if not F._is_tensor_image(img):
+        raise TypeError('tensor is not a torch image.')
+
+    mean = torch.mean(_rgb_to_grayscale(img))
+    mean_img = mean * torch.ones(img.shape, device=img.device)
+
+    return _blend(img, mean_img, contrast_factor)
+
+
+def adjust_saturation(img, saturation_factor):
+    """Adjust color saturation of an RGB image.
+
+    Args:
+        img (Tensor): Image to be adjusted.
+        saturation_factor (float):  How much to adjust the saturation. 0 will
+            give a black and white image, 1 will give the original image while
+            2 will enhance the saturation by a factor of 2.
+
+    Returns:
+        Tensor: Saturation adjusted image.
+    """
+    if not F._is_tensor_image(img):
+        raise TypeError('tensor is not a torch image.')
+
+    num_channels = img.shape[0]
+    gray_img = _rgb_to_grayscale(img).repeat(num_channels, 1, 1)
+
+    return _blend(img, gray_img, saturation_factor)
+
+
+def _blend(img1, img2, ratio):
+    return (ratio * img1 + (1 - ratio) * img2).clamp(0, 1)
+
+
+def _rgb_to_grayscale(img):
+    # ITU-R 601-2 luma transform, as used in PIL.
+    return 0.2989 * img[0] + 0.5870 * img[1] + 0.1140 * img[2]
```

Comment:
I think it would be great if we could preserve the `dtype` of the input here.
Could you do something like `gray.to(img.dtype)`? The good thing about `.to` is that it is a no-op if the `dtype` of `gray` is the same as the `dtype` passed.

Reply:
Done.